src/utils/performance_monitor.py

```
@@ -0,0 +1,547 @@
+                    recommendations=[
+                        "Review query processing",
+                        "Improve document indexing",
+                        "Check embedding model performance",
+                    ],
+                )
+            )
+
+        # Add alerts to history
+        for alert in alerts:
+            self.alerts.append(alert)
+            logger.warning(f"Performance Alert [{alert.severity}]: {alert.message}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)